execution-engine/src/native_modules/darknet_inference.rs

```diff
@@ -0,0 +1,202 @@
+//! A native module for ML inference on Darknet.
+//! Takes an input image, feeds it to the model and outputs a list of detected
+//! objects.
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE_MIT.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+use crate::fs::{FileSystem, FileSystemResult};
+use crate::native_modules::common::Service;
+use darknet::{BBox, Detection, Image, Network};
+use lazy_static::*;
+use serde::Deserialize;
+use std::cmp::Ordering;
+use std::env::{current_dir, set_current_dir};
+use std::fmt::Write as _;
+use std::fs::{create_dir, create_dir_all, read_to_string, remove_dir_all, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use wasi_types::ErrNo;
+
+lazy_static! {
+    // Current directory before moving to the module's directory
+    static ref WORK_DIR: PathBuf = current_dir().unwrap_or(PathBuf::from(""));
+
+    // The native module executes in a dedicated directory to mitigate data
+    // leaks
+    static ref MODULE_DIR: PathBuf = WORK_DIR.clone().join("darknet_inference_service");
```

Comment:
As I understand it, `MODULE_DIR` will be set to `WORK_DIR`. and `WORK_DIR` is being set to `current_dir()` or "". In what scenarios is `MODULE_DIR` different from the `CURRENT_DIR` and thus actually executes the module in a dedicated directory?

Reply:
`MODULE_DIR` will always be different from `WORK_DIR` since it has a suffix.
But if `current_dir()` fails, `MODULE_DIR` won't be an absolute path and subsequent fs operations will fail. That's not great but we can't return errors in `lazy_static`